Training_Preparation.md

```diff
@@ -0,0 +1,275 @@
+
+
+Let's take a closer look at this section. Within this JSON object, there is also an "endpoints" array. The entries "href" and "subprotocolBody" are particularly interesting. Evaluating this information allows us to know where and how we can access the submodel described in this shell descriptor.
+
+
+We want to spare you the trouble of copying these data individually and transferring them to other request bodies.
+
+
+If we now look at the current request body, we see that we use the string "urn:samm:io.catenax.part_type_information:1.0.0#PartTypeInformation" as a filter criterion to start another catalog query.
+
+#### i) Negotiation for a Contract over the PartType Asset
+
+
+Business as usual :)
+
+
+#### j) Querying the Negotiation Status
+As a response, we receive the contract ID.
+
+
+#### k) Requesting a Pull Transfer
+We provide the contract ID.
+
+
+#### l) Requesting an EDR
+As above.
+
+
+#### m) Pull Data from PartType-API
+
+
+We now see what information the PartType-API provides regarding this material. It's also interesting to note that the PURIS backend internally handled the response. If we take a quick look at the logs of our PURIS backend containers, we can see that the PartTypeInformationController has processed a request.
+
+
+### Task 4: Creating a New Sample Asset on Our EDC
+
+
+#### a) Asset Registration
+
+
+With this request, we create a new asset on the EDC of the supplier partner.
+
+
+Let's briefly look at the request body. We create a new fictitious API asset there, which we have given the identifier "urn:samm:io.catenax.sample_information:1.0.0#SampleInformation".
+
+
+Since we do not have an endpoint in the PURIS backend that can serve this asset, we simply use any URL in the "baseUrl" field that returns a list of artificial user names.
+
+
+#### b) Creating a Contract Definition
+
+
+Here, we reference the newly created asset in the request body.
+
+
+Additionally, we reference in the "accessPolicy" and "contractPolicy" fields two policies that our PURIS backend already created on the EDC during its boot process.
+
+
+### Task 5: Inserting the Sample Asset API into a Shell Descriptor
+
+
+#### a) Request a token:
+You know the drill ;)
+
+
+#### b) Lookup for a specific Shell Descriptor:
+As before.
+
+
+#### c) Querying a Specific Shell Descriptor:
+
+
+If we briefly look at the response, we will of course notice that our SampleAsset API is not included there.
+
+
+#### d) Modifying an Existing Shell Descriptor:
+
+
+Let's first calmly look at the request body. We have already manually inserted an entry for our SampleAsset API there. With this PUT request, we ensure that it can now be found for the material.
+
+
+#### e) Querying a Specific Shell Descriptor Again
+
+
+Let's briefly check if the previous operation was successful.
+
+
+#### f) Task for Fast Completers:
+Redo the steps from Task 3 a to g. Do you also find our SampleAsset API?
+
+
+### Task 6: Testing Our Sample Asset API
+
```

Comment:
Add some support: As a customer we now are going to retrieve the data added on supplier side in the previous task.